src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,100 @@
+import elasticsearch from 'elasticsearch';
+import { bindKey, partial, get, set, isEmpty, cloneDeep  } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Promise from 'bluebird';
+import Boom from 'boom';
+
+import createClient from './create_client';
+import filterHeaders from './filter_headers';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    this._client = this.createClient();
+    this.callAsKibanaUser = this.callAsKibanaUserFactory(this._client);
+
+    this._noAuthClient = this.createClient({ auth: false });
+    this.callWithRequest = this.callWithRequestFactory(this._noAuthClient);
+
+    return this;
+  }
+
+  /**
+   * callAsKibanaUser
+   *
+   * Makes a call to ES using the credentials in kibana.yml
+   *
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
+   * @returns {Promise}
+   */
+  callAsKibanaUserFactory(client) {
```

Comment:
That's a good point. We are currently utilizing it in X-Pack but we should be able to go about creating the client with `elasticsearch-js` plugins another way.